bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/api/esp32/WlanThermoEsp32Handler.java

```diff
@@ -0,0 +1,222 @@
+            data = gson.fromJson(json, Data.class);
+            logger.debug("Received at /data: {}", json);
+            json = httpClient.GET(config.getUri("/settings")).getContentAsString();
+            settings = gson.fromJson(json, Settings.class);
+            logger.debug("Received at /settings: {}", json);
+
+            // Update Channels if required
+            Map<String, String> properties = editProperties();
+            Boolean pmEnabled = settings.getFeatures().getBluetooth();
+            int pmChannels = pmEnabled ? data.getPitmaster().getPm().size() : 0;
+            int tempChannels = data.getChannel().size();
+
+            // Update properties
+            properties.put(WlanThermoBindingConstants.PROPERTY_MODEL, settings.getDevice().getDevice());
+            properties.put(WlanThermoBindingConstants.PROPERTY_SERIAL, settings.getDevice().getSerial());
+            properties.put(WlanThermoBindingConstants.PROPERTY_ESP32_BT_ENABLED,
+                    settings.getFeatures().getBluetooth().toString());
+            properties.put(WlanThermoBindingConstants.PROPERTY_ESP32_PM_ENABLED, pmEnabled.toString());
+            properties.put(WlanThermoBindingConstants.PROPERTY_ESP32_TEMP_CHANNELS, String.valueOf(tempChannels));
+            properties.put(WlanThermoBindingConstants.PROPERTY_ESP32_PM_CHANNELS, String.valueOf(pmChannels));
+            updateProperties(properties);
+
+            // Update channel state
+            for (Channel channel : thing.getChannels()) {
+                State state = wlanThermoEsp32CommandHandler.getState(channel.getUID(), data, settings);
+                if (state != null) {
+                    updateState(channel.getUID(), state);
+                } else {
+                    // if we could not obtain a state, try trigger instead
+                    String trigger = wlanThermoEsp32CommandHandler.getTrigger(channel.getUID(), data);
+                    if (trigger != null) {
+                        triggerChannel(channel.getUID(), trigger);
+                    }
+                }
+            }
+        } catch (URISyntaxException | InterruptedException | ExecutionException | TimeoutException e) {
+            logger.debug("Update failed, checking connection", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Update failed, reconnecting...");
+            ScheduledFuture<?> oldScheduler = pollingScheduler;
+            if (oldScheduler != null) {
+                oldScheduler.cancel(false);
+            }
+            for (Channel channel : thing.getChannels()) {
+                updateState(channel.getUID(), UnDefType.UNDEF);
+            }
+            checkConnection();
+        }
+    }
+
+    private void push() {
+        data.getChannel().forEach(c -> {
+            try {
+                String json = gson.toJson(c);
+                logger.debug("Pushing: {}", json);
```

Comment:
This is usually the most crucial part in case sth goes wrong. 
I'd prefer to leave it with a dedicated debug log so that users can easily include it in in bug reports.

Reply:
Is there any chance to log this only if something goes wrong?

Reply:
Payload is now only logged if response code != 200